types/event_record.go

```diff
@@ -162,10 +252,34 @@ func (e EventRecordsRaw) DecodeEventRecords(m *Metadata, t interface{}) error {
 
 		// create a pointer to with the correct type that will hold the decoded event
 		holder := reflect.New(field.Type().Elem())
-		err = decoder.Decode(holder.Interface())
-		if err != nil {
-			return fmt.Errorf("unable to decode event #%v with EventID %v, field %v_%v: %v", i, id, moduleName,
-				eventName, err)
+
+		// ensure first field is for Phase, last field is for Topics
+		numFields := holder.Elem().NumField()
+		if numFields < 2 {
```

Comment:
These are just sanity checks right? In the case of such an error triggered that would mean that the substrate version is not compatible with this GSRPC version, missed some record type or something core changed. Is there any other reasons this error would be triggered?

Reply:
Almost correct: 

Since the user is specifying `EventRecords`, this error happens if the user has included event record types events in the struct that do not include at least `Phase` and `[]Hash` for topics.

I added these error messages to improve developer experience.